### PR TITLE
Journalctl fields

### PR DESCRIPTION
**Pull request by kaarposoft**
resolved: ENABLE_DEBUG_HASHMAP needs pthread.h
utf8: add utf8_escape_non_printable_newline

loginctl: gained --fields option
**loginctl --fields:**
- print field names (with number of entries and number of values)
- with --fields --fields print field values
- with --fields=<FIELD> print values for this field

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -4013,6 +4013,9 @@ journalctl_LDADD += \
 	$(QRENCODE_LIBS)
 endif
 
+journalctl_SOURCES += \
+	src/journal/journal-fields.c
```
Comment (evverx) [inline Makefile.am]: There is a missing `src/journal/journal-fields.h`.
See https://semaphoreci.com/systemd/systemd/branches/pull-request-2159/builds/1

```
../src/journal/journalctl.c:59:28: fatal error: journal-fields.h: No such file or directory
 #include "journal-fields.h"
                            ^
compilation terminated.
../src/journal/journal-fields.c:43:28: fatal error: journal-fields.h: No such file or directory
 #include "journal-fields.h"
                            ^
compilation terminated.
```

**Comment by kaarposoft** [comment]
The file src/journal/journal-fields.h **is** included in the commit:
https://github.com/systemd/systemd/pull/2159/files#r47463495
https://github.com/kaarposoft/systemd/blob/journalctl_fields/src/journal/journal-fields.h

There are two semaphoreci "threads". One has failed, the other one passed. I am not sure what the difference between the two threads is.

**Comment by evverx** [comment]
> The file src/journal/journal-fields.h is included in the commit

Yeah.
I mean, there is no `src/journal/journal-fields.h` in the `Makefile.am`

> I am not sure what the difference between the two threads is.

`make check` passed.
`make distcheck` failed.

**Comment by zonque** [comment]
Your PR now contains a back-merge of master, which shouldn't be there. Could you rebase?

One of the thread of Semaphore boots up a Qemu container with the systemd version built from the branch. And this one failed due to a crash during shutdown. If that's a false positive, we should keep an eye on it. But the fact that this didn't happen before makes be believe it's not.

**Comment by kaarposoft** [comment]
I am sorry for the mess - seems that I am making all the mistakes in the book )-:
Thanks for staying with me.

As for the original distcheck failure:
I was verifying locally with just a "make", not "make distcheck".
It seems that the missing .h file in Makefile.am is not a problem 
for a standard "make", but the ".tar" file created during
"make distcheck" will miss the file if it is not in Makefile.am.
So I have now added "src/journal/journal-fields.h" to "Makefile.am"

I am always trying to keep up with master, hence the back-merge of master.
My apologies, I did not know that this was not wanted.
I am not a git expert, but I think that rebasing code already
committed upstream will be a mess.
So I have created a new pull request on a new branch instead:
https://github.com/systemd/systemd/pull/2167

As for the crash of the Qemu container I am at a loss.
No matter how much I borked the journalctl update,
I think that it should not be able to trigger a Qemu crash.

**Pull request closed without merging**